src/id/doujindesu/src/eu/kanade/tachiyomi/extension/id/doujindesu/DoujinDesu.kt

```diff
@@ -368,29 +377,56 @@ class DoujinDesu : ParsedHttpSource(), ConfigurableSource {
 
     override fun mangaDetailsParse(document: Document): SManga {
         val infoElement = document.select("section.metadata").first()!!
+        val AuthorName = when {
+            infoElement.select("td:contains(Author) ~ td")
+                .isEmpty() -> "Tidak Diketahui"
+            else -> infoElement.select("td:contains(Author) ~ td")
+                .joinToString { it.text() }
+        }
+        val GroupName = when {
+            infoElement.select("td:contains(Group) ~ td")
+                .isEmpty() -> "Tidak Diketahui"
+            else -> infoElement.select("td:contains(Group) ~ td")
+                .joinToString { it.text() }
+        }
+        val AuthorParser = when {
+            AuthorName.isNullOrEmpty() || AuthorName == "Tidak Diketahui" -> GroupName.takeIf { !it.isNullOrEmpty() && it != "Tidak Diketahui" } ?: "Tidak Diketahui"
```

Comment:
'Tidak Diketahui' should not be used in the output for native fields such as `author`. It should either be null or empty string (whatever the app uses to detect a missing value) so it is up to the app to pick an appropriate string.

You can still use 'Tidak Diketahui' for other fields you only put in the description.

Reply:
Like this?
```kotlin
val AuthorName = when { 
    infoElement.select("td:contains(Author) ~ td")
.isEmpty() -> null 
    else -> infoElement.select("td:contains(Author) ~ td").joinToString { it.text() } 
} 

val AuthorParser = when { 
    AuthorName.isNullOrEmpty() && GroupName.isNullOrEmpty() -> null 
    AuthorName.isNullOrEmpty() -> GroupName 
    else -> AuthorName 
}
```